drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/ValidatorTest.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.guided.dtable.client.widget;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.drools.workbench.models.guided.dtable.shared.model.BaseColumn;
+import org.drools.workbench.models.guided.dtable.shared.model.CompositeColumn;
+import org.drools.workbench.models.guided.dtable.shared.model.ConditionCol52;
+import org.drools.workbench.models.guided.dtable.shared.model.DTCellValue52;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class ValidatorTest {
+
+    @Test
+    public void testIsConditionValidWhenHeaderIsBlank() throws Exception {
+        final Validator validator = new Validator();
+
+        assertFalse(validator.isConditionValid(fakeConditionCol52("")));
+    }
+
+    @Test
+    public void testIsConditionValidWhenHeaderIsNotUnique() throws Exception {
+        final ConditionCol52 conditionCol1 = fakeConditionCol52("header");
+        final ConditionCol52 conditionCol2 = fakeConditionCol52("header");
+        final Validator validator = new Validator(fakeCompositeColumn(conditionCol1,
+                                                                      conditionCol2));
+
+        assertFalse(validator.isConditionValid(conditionCol2));
+    }
+
+    @Test
+    public void testIsConditionValidWhenHeaderIsValid() throws Exception {
+        final ConditionCol52 conditionCol1 = fakeConditionCol52("header1");
+        final ConditionCol52 conditionCol2 = fakeConditionCol52("header2");
+        final Validator validator = new Validator(fakeCompositeColumn(conditionCol1,
+                                                                      conditionCol2));
+
+        assertFalse(validator.isConditionValid(conditionCol2));
```

Comment:
In the context of GUVNOR-2821 I would expect `assertTrue` here, can you please update the `fakeConditionCol52` method to achieve this? Let me know if you don't see my point.